.github/workflows/dotnet-build.yml

```diff
@@ -0,0 +1,29 @@
+name: .NET Build
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+jobs:
+  build:
```

Comment:
We need to figure out a way to extract a more useful name out of the action. When selecting the status checks in the branch protection I see the following:
![image](https://user-images.githubusercontent.com/1426795/195176806-32867e2b-1084-4840-9eb4-f62b2a819f61.png)

That's not super helpful for knowing what I'll be selecting. 

See how the old Circle-ci used a different name (Ex: `build-mono`)

Reply:
Hmm, okay - that's just the job name within the workflow (`build`) and a list of "matrix parameters" used in generating job variations (in this case, build configurations). I can change the job name to anything though, like `dotnet-build` and `mono-build`?

Reply:
Sounds good. Maybe reverse the terms like we had before? My thought is that the actions build, test, etc.. are more likely to fail together vs specific compilers failing. So the failures will tend to show grouped in the UI? (IE: Failures next to one another)

Also, more likely that a specific action isn't supported by all platforms so it's better to group by action?
Ex:
 - build-dotnet
 - build-mono
 - test-dotnet
 - test-dotnet
 - profile-dotnet

In this example I'm assuming there is no headless profiling support from Mono.

Reply:
The list of tests on a PR seems to have arbitrary order, I guess whichever ones start running first since they're all done in parallel, but maybe I'm not seeing a context where they're sorted.

Either way, I have nothing against reversing the naming, so `build-dotnet`/`build-mono` it is!